src/main/java/org/htmlunit/javascript/host/Location.java

```diff
@@ -226,11 +227,13 @@ public void reload(final boolean force) throws IOException {
         final HtmlPage htmlPage = (HtmlPage) webWindow.getEnclosedPage();
         final WebRequest request = htmlPage.getWebResponse().getWebRequest();
 
+        // update request url with location.href in case hash was changed
+        request.setUrl(new URL(getHref()));
```

Comment:
i fear setting this here will change the getUrl() response because the getUrl() method reads from the request

Reply:
@rbri added 1 more commit to update the tests. Could you take a look at it ?